RHOAI-16971 LAB-tuning - user (#775)
opendatahub-io/opendatahub-documentation: modules/deploying-teacher-and-judge-models.adoc

:_module-type: PROCEDURE

[id="deploying-teacher-and-judge-models_{context}"]
= Deploying teacher and judge models

[role='_abstract']

> To use LAB-tuning in {productname-short}, you must deploy both a teacher model and a judge model. The teacher model generates synthetic data, and the judge model evaluates the performance of the LAB-tuned model. 

Reviewer: Not sure if you'll like this test but I think it would be good to include the names of the required models and also the possibility to deploy them in another cluster

```suggestion
LAB-tuning in {productname-short} uses a teacher model (link:https://catalog.redhat.com/software/containers/rhelai1/modelcar-mixtral-8x7b-instruct-v0-1/67922f1e167e94db874af7ab[rhelai1/modelcar-mixtral-8x7b-instruct-v0-1:1.4]) to generate synthetic data and a judge model (link:https://catalog.redhat.com/software/containers/rhelai1/modelcar-prometheus-8x7b-v2-0/67922f21a4baf873b6f43d8c[rhelai1/modelcar-prometheus-8x7b-v2-0:1.4]) to evaluate the performance of the LAB-tuned model. 

The models can be deployed in the same cluster where the LAB-tuning process is going to be running or in a different one, providing the required endpoint and authentication information. 
```

.Prerequisites

* You have logged in to {productname-long}.

> * Your cluster administrator has configured LAB-tuning in your cluster, as described in _Enabling LAB-tuning_. 

Reviewer: ```suggestion
* Your cluster administrator has configured LAB-tuning in your cluster, as described in _Enabling LAB-tuning_. 
* Your cluster has available GPUs capable to deploy the teacher and judge models.
```

* For the judge model (link:https://catalog.redhat.com/software/containers/rhelai1/modelcar-prometheus-8x7b-v2-0/67922f21a4baf873b6f43d8c[rhelai1/modelcar-prometheus-8x7b-v2-0:1.4]), the {openshift-platform} cluster has a worker node with 1 or multiple GPUs capable of running this model (for example, a2-ultragpu-2g, which contains 2 x NVIDIA A100 with 80GB vRAM) and has at least at least 100 GiB of available disk storage to store the model.
* You have enabled the LAB-tuning features in your current session, as described in _Making LAB-tuning features visible_.
* You have created a data science project and configured its pipeline server.

.Procedure
. From the {productname-short} dashboard, click *Models* -> *Model catalog*.
+
The *Model catalog* page opens showing the pre-built, curated models that are available to your organization.
. Under *{org-name} models*, select a model with the `LAB teacher` label.

> . On the model details page, click the *Deploy model* button to deploy a model server that hosts the selected model and makes it available to your project.

Reviewer 2: ```suggestion
. On the model details page, click *Deploy model* to deploy a model server that hosts the selected model and makes it available to your project.
```
